# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/commonc++/hotfix-gcc16.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Move getline default arguments out of the friend declaration, rejected by
current g++, include sys/stat.h for mkfifo.

--- a/inc/cc++/string.h	2026-09-25 19:43:00.336090480 +0200
+++ b/inc/cc++/string.h	2026-09-25 19:43:00.337705033 +0200
@@ -57,6 +57,9 @@ namespace ost {
 #endif
 
 class MemPager;
+class String;
+
+__EXPORT std::istream &getline(std::istream &is, String &str, char delim = '\n', size_t size = 0);
 
 /**
  * This is a generic and portable string class.  It uses optimized
@@ -731,7 +734,7 @@ public:
      * @param delim deliminator to use.
      * @param size optional size limitor.
      */
-    friend __EXPORT std::istream &getline(std::istream &is, String &str, char delim = '\n', size_t size = 0);
+    friend __EXPORT std::istream &getline(std::istream &is, String &str, char delim, size_t size);
 
     /**
      * Stream the content of our string variable directly to a C++
--- a/src/applog.cpp	2026-09-25 19:43:55.647105331 +0200
+++ b/src/applog.cpp	2026-09-25 19:43:55.661585587 +0200
@@ -45,6 +45,7 @@
 #include <cstdlib>
 #include <stdarg.h>
 #include <errno.h>
+#include <sys/stat.h>
 
 // TODO sc: test if has to move up now that it is into commoncpp
 // NOTE: the order of inclusion is important do not move following include line
